pkg/controller/catalog/storage/mem_storage_instance.go

```diff
@@ -0,0 +1,63 @@
+package storage
+
+import (
+	"errors"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+)
+
+var (
+	errInstanceAlreadyExists = errors.New("instance already exists")
+	errNoSuchInstance        = errors.New("no such instance")
+)
+
+type memStorageInstance struct {
+	// maps instance ID to instance
+	instances map[string]*servicecatalog.Instance
+}
+
+func newMemStorageInstance() *memStorageInstance {
+	return &memStorageInstance{instances: make(map[string]*servicecatalog.Instance)}
+}
+
+func (m *memStorageInstance) List() ([]*servicecatalog.Instance, error) {
+	ret := make([]*servicecatalog.Instance, len(m.instances))
+	i := 0
+	for _, val := range m.instances {
+		ret[i] = val
+		i++
+	}
+	return ret, nil
+}
+
+func (m *memStorageInstance) Get(name string) (*servicecatalog.Instance, error) {
+	ret, ok := m.instances[name]
+	if !ok {
+		return nil, errNoSuchInstance
+	}
+	return ret, nil
```

Comment:
In such a scenario, yes, I agree. Do we have some inclination of how likely that is? Personally, I see very limited value to any storage where the contents of the store evaporate when a single process (e.g. the catalog controller) dies. i.e. I don't foresee this being used for anything but testing. _Maybe_ the right thing to do here is to rename all the "memory storage" stuffs as "fake storage?" @arschles any opinions on that?